Xplat/src/main/java/vazkii/botania/common/entity/ManaSparkEntity.java

```diff
@@ -59,7 +59,11 @@ public class ManaSparkEntity extends SparkBaseEntity implements ManaSpark {
 
 	private final Set<ManaSpark> transfers = Collections.newSetFromMap(new WeakHashMap<>());
 
-	private int removeTransferants = 2;
+	private final ArrayList<ManaSpark> transfersTowardsSelfToRegister = new ArrayList<>();
```

Comment:
Any risk of sparks hanging around in this list and leaking memory? I'd either use weak references or filter out `remove`d sparks every tick.